dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLitmusC.java

```diff
@@ -252,7 +288,14 @@ public Expression visitReAtomicFetchOp(LitmusCParser.ReAtomicFetchOpContext ctx)
     public Expression visitC11AtomicOp(LitmusCParser.C11AtomicOpContext ctx) {
         Register register = getReturnRegister(true);
         Expression value = returnExpressionOrOne(ctx.value);
-        Event event = EventFactory.Atomic.newFetchOp(register, getAddress(ctx.address), value, ctx.op, ctx.c11Mo().mo);
+        Expression address = getAddress(ctx.address);
+        Event event = EventFactory.Atomic.newFetchOp(register, address, value, ctx.op, ctx.c11Mo().mo);
+        addMemoryObjectTags(event, address);
+        if (ctx.openCLScope() != null) {
+            event.addTags(ctx.openCLScope().scope);
+        } else {
+            event.addTags(Tag.OpenCL.DEFAULT_SCOPE);
+        }
```

Comment:
Does this mean regular C11 litmus code always comes with OpenCL tagging? I don't think this will cause issues, but I'm wondering if the parser should already be aware of the target architecture to prevent unnecessary tagging.

Reply:
See @hernanponcedeleon comment about the OpenCL header to distinguish between OpenCL and pure C11.